### ISSUE TITLE
Ship a real JS library that allow you to work with data more easily

### ISSUE THREAD

**Issue by kumar303** [mozilla/payments-config#4]
Shipping JSON doesn't really make sense. We should ship a JS module that allows you to do things like:

```
var config = require('mozilla-payments-config');
config.products.forEach(function(product) {
  // product.id ...
})
```

**Comment by andymckay-limited-access** [comment]
Would it make sense if we made payments-service an API end point for serving out the JSON from payments-config?

So payments-ui would hit a well known end point like http://pay.dev:8000/api/config/mozilla-concrete-brick.

As @muffinresearch said, the amount of products will hopefully grow and grow so that shipping all the products won't make sense from a size point of view.

**Comment by andymckay-limited-access** [comment]
Also, it looks like in your example, you are suggesting we iterate over the products. Is that a use case we need?

**Comment by kumar303** [comment]
> Also, it looks like in your example, you are suggesting we iterate over the products. Is that a use case we need?

Yes it is, but my example code was wrong. What I meant was to show looping over all of a seller's products. That is, sellers might need to create a page showing all of their products.

**Comment by kumar303** [comment]
> Would it make sense if we made payments-service an API end point for serving out the JSON from payments-config?

In the case where we have lots of sellers and lots of products then yeah this would make sense. If we only have a few sellers and products it might still make sense to bake it all into the library. It will be a tradeoff: hitting the API is expensive but bundling more JS data into the client is also costly.

**Comment by andymckay** [comment]
If sellers want to create a page looping over their products, they can do so. I don't think we should worry about creating a use case for that need, that way ends a rabbit nest of a catalog and we certainly don't need that now.

**Comment by andymckay** [comment]
Agreed on the trade off, for the moment we'll have very few, so I say optimise for that.
